src/TypeSpec.Extension/Emitter.Csharp/src/lib/operation.ts

```diff
@@ -259,7 +259,7 @@ export function loadOperation(
                     : InputOperationParameterKind.Client
                 : InputOperationParameterKind.Method;
         return {
-            Name: param.name,
+            Name: getProjectedNameForCsharp(sdkContext, param) ?? param.name,
```

Comment:
As this is inside c# emitter, do we need to include csharp in function name?

Reply:
I am wondering whether we could also put ' ?? param.name' inside function body and just call it getTypeName()

Reply:
Different Type (Operation, Model, ModelProperty) may have different name logic, some may have friendlyName, or some may get the Name to combine the template args. And it is in a order, such as if projectedName is undefined, and try to get FriendlyName, if also undefined, then it origin name.
So we cannot combine all those in `getTypeName` and we need to just handle projectedName and return undefined if it is not defined.